src/components/ZetkinCalendar/index.tsx

```diff
@@ -51,10 +72,42 @@ const ZetkinCalendar = ({ baseHref, events, campaigns , tasks }: ZetkinCalendarP
         }
     };
 
+    const isTodayBeforeView = () =>{
+        if (focusDate.getFullYear() > dayjs().get('year')) return true;
+        if (focusDate.getFullYear() < dayjs().get('year')) return false;
+        //only needed if same year
+        if (range === CALENDAR_RANGES.MONTH) {
+            if (focusDate.getMonth() > dayjs().get('month')) return true;
+        }
+        else if (range === CALENDAR_RANGES.WEEK) {
+            if (focusDate.getWeekNumber() > dayjs().week()) return true;
+        }
+        return false;
+    };
```

Comment:
I feel like this is unnecessarily complex, and also doesn't behave the way I would expect.

If today's is last of July, it will be visible in the month view UI even if the currently focused month is August (because August starts on a Sunday and displays the full preceding week of July as well). But this function will return `false`. I would not have considered that desired behavior. But that's my personal opinion and it could very well be argued that it is exactly how it should work.

A less complex way IMO would be to check whether `firstDateInView < today < lastDateInView`. The `firstDateInView` and `lastDateInView` are already calculated in the week and month views and that logic could be moved to utility functions that are reused here.

But it's working, and the code is good, so lets keep it like this for now. No changes needed!

Reply:
Is `dayjs().week()` just getting the current week number? 
If so, I wouldn't mix it with the polyfill `.getWeekNumber()` -- as discussed in https://github.com/zetkin/app.zetkin.org/issues/329 there's a bug in this, and it will return the wrong week from 2022 onwards. 

I'd use the polyfill twice, or dayJS twice.

I also agree w @richardolsson that is todayInView() should consider the entire calendar view, not just the month start and end. Otherwise we will end up with two blue dots on the page when today is before the current month, but inside the calendar view, which would be weird.

Reply:
See 9dc6847113d5e2215abe6aa2e2d89f3d295a2558 for simpler solution.